packages/flutter_tools/lib/src/windows/visual_studio_project.dart

```diff
@@ -0,0 +1,64 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:xml/xml.dart' as xml;
+
+import '../base/file_system.dart';
+import '../globals.dart' as globals;
+
+/// A utility class for interacting with Visual Studio project files (e.g.,
+/// .vcxproj).
+class VisualStudioProject {
+  /// Creates a project object from the project file at [file].
+  VisualStudioProject(this.file) {
+    try {
+      content = xml.parse(file.readAsStringSync());
+    } on xml.XmlParserException {
+      // Silently continue; formatUnderstood will return false.
+    }
+  }
+
+  /// The file corresponding to this object.
+  final File file;
+
+  /// The content of the project file.
+  xml.XmlDocument content;
+
+  /// Whether or not the the project file was correctly parsed.
+  ///
+  /// If false, this could indicate that the project file is damaged, or that
+  /// it's an unsupported project type.
+  bool get formatUnderstood => content != null;
+
+  /// Returns the ProjectGuid for the project, or null if it's not present.
+  String get guid {
```

Comment:
I would cache the `guid` and `name` fields and lazily calculate them on demand. The usual pattern is something like:

```
String _field;
String get field {
  return _filed ??= _computeField();
}
```

Reply:
Done. (The guid caching won't actually cache on failure, but failure should only happen if someone has done something truly bizarre to their project file so I didn't think it was worth trying to optimize that failure case.)